cou/apps/core.py

```diff
@@ -422,7 +431,7 @@ def generate_upgrade_plan(self, target: str) -> UpgradeStep:
         for step in all_steps:
             if step:
                 upgrade_steps.add_step(step)
-        return upgrade_steps
+        return upgrade_steps if upgrade_steps.sub_steps else None
```

Comment:
What if we provide `empty` property, and basically skip any empty step instead of returning the `None`?